src/chainrules.jl

```diff
@@ -6,3 +6,122 @@ ChainRulesCore.@scalar_rule(
     ),
     (x, - tanh(Ω + b) * x, x - 1),
 )
+
+# `OrderedBijector`
+function ChainRulesCore.rrule(::typeof(_transform_ordered), y::AbstractVector)
+    function _transform_ordered_adjoint(Δ)
+        Δ_new = similar(y)
+        n = length(Δ)
+        @assert n == length(Δ_new)
+
+        s = sum(Δ)
+        Δ_new[1] = s
+        @inbounds for i in 2:n
+            # Equivalent to
+            #
+            #    Δ_new[i] = sum(Δ[i:end]) * yexp[i - 1]
+            #
+            s -= Δ[i - 1]
+            Δ_new[i] = s * exp(y[i])
+        end
+
+        return (ChainRulesCore.NoTangent(), Δ_new)
+    end
+
+    return _transform_ordered(y), _transform_ordered_adjoint
+end
+
+function ChainRulesCore.rrule(::typeof(_transform_ordered), y::AbstractMatrix)
+    function _transform_ordered_adjoint(Δ)
+        Δ_new = similar(y)
+        n = length(Δ)
+        @assert n == size(Δ_new, 1)
+
+        s = sum(Δ)
+        Δ_new[1, :] .= s
+        @inbounds for i in 2:n
+            # Equivalent to
+            #
+            #    Δ_new[i] = sum(Δ[i:end]) * yexp[i - 1]
+            #
+            s -= Δ[i - 1]
+            Δ_new[i, :] = s * exp.(y[i, :])
+        end
+
+        return (ChainRulesCore.NoTangent(), Δ_new)
+    end
+
+    return _transform_ordered(y), _transform_ordered_adjoint
+end
+
+function ChainRulesCore.rrule(::typeof(_transform_inverse_ordered), x::AbstractVector)
+    r = similar(x)
+    @inbounds for i = 1:length(r)
+        if i == 1
+            r[i] = 1
+        else
+            r[i] = x[i] - x[i - 1]
+        end
+    end
+
+    function _transform_inverse_ordered_adjoint(Δ)
+        Δ_new = similar(x)
+        @assert length(Δ_new) == length(Δ)
+
+        n = length(Δ_new)
+        @inbounds for j = 1:n - 1
+            Δ_new[j] = (Δ[j] / r[j]) - (Δ[j + 1] / r[j + 1])
+        end
+        @inbounds Δ_new[n] = Δ[n] / r[n]
+
+        return (ChainRulesCore.NoTangent(), Δ_new)
+    end
+
+    y = similar(x)
+    @inbounds y[1] = x[1]
+    @inbounds for i = 2:length(x)
+        y[i] = log(r[i])
+    end
+
+    return y, _transform_inverse_ordered_adjoint
+end
+
+function ChainRulesCore.rrule(::typeof(_transform_inverse_ordered), x::AbstractMatrix)
+    r = similar(x)
+    @inbounds for j = 1:size(x, 2), i = 1:size(x, 1)
+        if i == 1
+            r[i, j] = 1
+        else
+            r[i, j] = x[i, j] - x[i - 1, j]
+        end
+    end
```

Comment:
Same here, maybe just `r = diff(x; dims=1)`?

Reply:
Same as above: IMO complicates the indexing and it doesn't do anything for perf (in a quick benchmark).